Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -0,0 +1,203 @@
+        self.cameraView = cameraView
+    }
+
+    deinit {
+        propertyAnimator.stopAnimation(false)
+        propertyAnimator.finishAnimation(at: .current)
+        cameraView.removeFromSuperview()
+    }
+
+    /// Starts the animation if this animator is in `inactive` state. Also used to resume a "paused" animation.
+    public func startAnimation() {
+
+        if self.state != .active {
+
+            guard let delegate = delegate else {
+                fatalError("CameraAnimator delegate cannot be nil when starting an animation")
+            }
+
+            guard let animation = animation else {
+                fatalError("Animation cannot be nil when starting an animation")
+            }
+
+            // Set up the short lived camera view
+            delegate.addViewToViewHeirarchy(cameraView)
+
+            var cameraTransition = CameraTransition(with: delegate.camera, initialAnchor: delegate.anchorAfterPadding())
+            animation(&cameraTransition)
+
+            propertyAnimator.addAnimations { [weak cameraView] in
+                guard let cameraView = cameraView else { return }
+                cameraView.syncLayer(to: cameraTransition.toCameraOptions) // Set up the "to" values for the interpolation
+            }
+
+            cameraView.syncLayer(to: cameraTransition.fromCameraOptions) // Set up the "from" values for the interpoloation
+            transition = cameraTransition // Store the mutated camera transition
+        }
+
+        propertyAnimator.startAnimation()
+    }
+
+    /// Starts the animation after a delay
+    /// - Parameter delay: Delay (in seconds) after which the animation should start
+    public func startAnimation(afterDelay delay: TimeInterval) {
+        delayedAnimationTimer = Timer.scheduledTimer(withTimeInterval: delay, repeats: false, block: { [weak self] (timer) in
+            guard let self = self else { return }
+            self.startAnimation()
+            timer.invalidate()
+        })
+    }
+
+    /// Pauses the animation.
+    public func pauseAnimation() {
+        propertyAnimator.pauseAnimation()
+    }
+
+    /// Stops the animation.
+    public func stopAnimation() {
+        propertyAnimator.stopAnimation(false)
+        propertyAnimator.finishAnimation(at: .current)
+    }
+
+    /// Add animations block to the animator.
+    internal func addAnimations(_ animations: @escaping (inout CameraTransition) -> Void) {
+        animation = animations
+    }
+
+    /// Add a completion block to the animator. 
+    public func addCompletion(_ completion: @escaping AnimationCompletion) {
+        let wrappedCompletion = wrapCompletion(completion)
+        propertyAnimator.addCompletion(wrappedCompletion)
+    }
+
+    internal func wrapCompletion(_ completion: @escaping AnimationCompletion) -> (UIViewAnimatingPosition) -> Void {
+        return { [weak self] animationPosition in
+            guard let self = self, let delegate = self.delegate else { return }
```

Comment:
Only some of the logic that follow depends on having a non-nil delegate. Should we still skip the other parts?

Reply:
It doesn't really make a difference functionally, since the animation can't start if there's no delegate. But worth cleaning up for sure.